distributed/diagnostics/status_monitor.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import print_function, division, absolute_import
+
+import json
+from tornado import gen
+
+from ..core import rpc
+from ..utils import ignoring, is_kernel
+from ..executor import default_executor
+from ..scheduler import Scheduler
+
+from tornado.httpclient import AsyncHTTPClient
+from tornado.ioloop import IOLoop, PeriodicCallback
+
+with ignoring(ImportError):
+    from bokeh.models import ColumnDataSource
+    from bokeh.models.widgets import DataTable, TableColumn
+    from bokeh.plotting import vplot, output_notebook, show
+    from bokeh.io import curstate, push_notebook
+
+class Status_Monitor(object):
+    """ Display the tasks running and waiting on each worker
+
+    Parameters
+    ----------
+    addr: tuple, optional
+        (ip, port) of scheduler.  Defaults to scheduler of recent Executor
+    interval: Number, optional
+        Interval between updates.  Defaults to 1s
+    """
+    def __init__(self, addr=None, interval=1000.00):
+        if addr is None:
+            scheduler = default_executor().scheduler
+            if isinstance(scheduler, rpc):
+                addr = (scheduler.ip, scheduler.port) # doesn't work
```

Comment:
Yeah, this port is for the TCP server, which we could also start using if performance becomes an issue.